src/hooks/useList.js

```diff
@@ -29,7 +30,23 @@ export default function useList() {
     visibilityUndoButton,
   }
 
-  function addListItem(title, isChecked = false) {
+  function handleInputChange(event) {
+    const fieldValue = event.target.value
+    const targetId = event.target.name
+
+    setList({
+      ...list,
+      byId: {
+        ...list.byId,
+        [targetId]: {
+          ...list.byId[targetId],
+          title: fieldValue,
+        },
+      },
+    })
+  }
+
+  function addListItem(title = '', isChecked = false) {
```

Comment:
You could think about basic validation f.e. not adding a new Item if it`s emtpy?
Right now i just start up the App, click on the Button once and then press enter 5 times and i have 6 empty Items in the List. There is no point in allowing that.